File: Java_Programs/Java_Lab_Program/Lab_3/Program_2.java
// Write a program to create basic calculator by getting 2 numbers and an operation(+,-
// ,/,*,%) from the user and apply the operation given in a string on the given numbers.

import java.util.Scanner;

public class Program_2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Input: two numbers
        System.out.print("Enter first number: ");
        double num1 = sc.nextDouble();

        System.out.print("Enter second number: ");
        double num2 = sc.nextDouble();

        // Input: operation
        System.out.print("Enter an operator (+, -, *, /, %): ");
        String op = sc.next();

        double result;

        // Perform operation based on user input
        switch (op) {
            case "+":
                result = num1 + num2;
                System.out.println("Result: " + result);
                break;

            case "-":
                result = num1 - num2;
                System.out.println("Result: " + result);
                break;

            case "*":
                result = num1 * num2;
                System.out.println("Result: " + result);
                break;

            case "/":
                if (num2 != 0)
                    result = num1 / num2;
                else {
                    System.out.println("Error: Division by zero!");
                    sc.close();
                    return;
                }
                System.out.println("Result: " + result);
                break;

            case "%":
                if (num2 != 0)
                    result = num1 % num2;
                else {
                    System.out.println("Error: Division by zero!");
                    sc.close();
                    return;
                }
                System.out.println("Result: " + result);
                break;

            default:
                System.out.println("Invalid operator!");
        }

        sc.close();
    }
}

File: C_DSA_Lab/LAB_4/Program_2.c
// Write a C program to swap two numbers, calling an UDF by value. 

#include<stdio.h>

void swap_val(int x,int y){
    x=x+y;
    y=x-y;
    x=x-y;

    printf("Value Of a : %d \n",x);
    printf("Value Of b : %d \n",y);
}

int main(){

    int a,b;

    printf("Enter Value Of a : ");
    scanf("%d",&a);

    printf("Enter Value Of b : ");
    scanf("%d",&b);

    printf("Value Of a : %d \n",a);
    printf("Value Of b : %d \n",b);

    printf("\n------------------------\n");
    
    swap_val(a,b);

    printf("\n------------------------\n");
    printf("Value Of a : %d \n",a);
    printf("Value Of b : %d \n",b);
    printf("\n------------------------\n");

}